crates/cxx-qt-lib/src/core/qtlogging.rs

```diff
@@ -36,6 +36,18 @@ mod ffi {
         /// Outputs a message in the Qt message handler.
         fn qt_message_output(msgType: QtMsgType, context: &QMessageLogContext, string: &QString);
 
+        /// Generates a formatted string out of the type, context, str arguments.
+        #[cxx_name = "qFormatLogMessage"]
+        fn format_log_message(
+            msgType: QtMsgType,
+            context: &QMessageLogContext,
+            string: &QString,
+        ) -> QString;
+
+        /// Generates a formatted string out of the type, context, str arguments.
+        #[cxx_name = "qSetMessagePattern"]
+        fn set_message_pattern(pattern: &QString);
```

Comment:
The Qt documentation states that qFormatLogMessage is thread-safe.

However, it does **not** state that `set_message_pattern` is thread-safe!
Can you investigate whether this function is thread-safe?
Otherwise we may need to make it `unsafe` or enforce safety somehow...

Reply:
Note that we're having similar issues with #1205

Reply:
I think it's fine to mark it as unsafe, I should probably add a comment explaining why it is right?